lib/client.js

```diff
@@ -75,8 +82,22 @@ function messageHandler(client, event) {
       pendingPromise.resolve(data.result);
     }
   } else if (ZAF_EVENT.test(data.key)) {
-    var key = data.key.replace(ZAF_EVENT, '');
-    if (client) {
+    var key = data.key.replace(ZAF_EVENT, ''),
+        msg = { appGuid: client._appGuid };
+    if (data.reply) {
+      if (client._repliesPending[key]) { return; }
+      msg.key = 'iframe.reply:' + key;
+      client._repliesPending[key] = true;
+      when(client._messageHandlers[key]).then(
+        rawPostMessage.bind(null, client, msg)
+      ).catch(function(reason) {
+        msg.error = true;
```

Comment:
could make this the same format as the error responses for promises, i.e. { error: { msg: 'reason' } } - although again, that would require a ZAF change.